app/models/veteran_onboarding.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# The VeteranOnboarding model represents the onboarding status of a veteran.
+# Each instance corresponds to a veteran who is in the process of onboarding.
+#
+# == Schema Information
+#
+# Table name: veteran_onboardings
+#
+#  id                      :bigint           not null, primary key
+#  icn                     :string           not null, unique
+#  display_onboarding_flow :boolean          default(TRUE)
+#  created_at              :datetime         not null
+#  updated_at              :datetime         not null
+#
+class VeteranOnboarding < ApplicationRecord
```

Comment:
Can you update your underlying schema so that ICN is no longer located here? If you want to associate with a veteran you should instead have an association with a `UserAccount` object, which stores an ICN.

Reply:
The reason I ask this is there is some upkeep to ICN that you don't want to have to do on your own, by associating with `UserAccount` you don't have to worry about any of that. Also, `UserAccounts` already deal with 'unverified' users and the possible upgrade to a 'verified' user, again, functionality you don't want to have to replicate

Reply:
Hey @bosawt ! Yes, totally reasonable.  I'll make the change.

Reply:
I have a new [PR](https://github.com/department-of-veterans-affairs/vets-api/pull/16626) up for that change.  Thanks for the heads up.